### ISSUE TITLE
Beginner's question

### ISSUE THREAD

**Issue by ZCG113035** [amforte/Topographic-Analysis-Kit#36]
Hello Adam, I am a beginner of TAK. I have no background in computer language learning. I have some basic questions to ask you: 1) I know about TAK's operation flow from the workflow. After doing Makestream, I still need to perform ConditionDEM And RemoveFlats and FindThreshold operation?
2) Can you provide tutorial steps similar to the Profiler Toolbar guide manual or the case code published in the Blog archive in TopoToolbox? In this case, it will be a double-dose effort for our beginners, which will allow us to quickly Analyze the study areas we are interested in.
Here I wish you a Merry Christmas in advance.

**Comment by amforte** [comment]
Hi @ZCG113035, for 1) no, ConditionDEM, RemoveFlats and FindThreshold are all optional for specific purposes. You can take the output of MakeStreams and it use it directly in other TAK functions.

For 2) I've been in the process of developing something like that, but I hope one can appreciate that I already have written a very large manual and that presenting case by case examples for ALL of the tools in TAK (Profiler was basically just my equivalent KsnProfiler) is a lot of effort (with me as the only developer) so it's taking a while.

**Comment by ZCG113035** [comment]
Thank you for your reply. Through this period of study, I can manually use most of the functions in TAK. TAK is very easy to use and powerful. Thank you for your contribution to this. I would like to ask how small the minarea should be. If it is 30 m pixels, it will usually be 10 ^ 6 in TAK, but it will be 10 ^ 3 in Topotoolbox, which is 0.9km ^ 2. I am currently dealing with the eastern Qinghai-Tibet Plateau Marginal DEM data, there is a deep-cut canyon area. The resolution of my DEM data is 145m. I also chose minarea as 10 ^ 6, and extracted the stream profile of interest based on manual selection.
I suspect it is because of the quality of the data, but I have a few questions: 1) How big should minarea be, is it based on the resolution of the DEM data? 2) When doing Sementprojector, how much should the reference concavity be 0.45 or 0.5 Choose between 0.4-0.6). There are several functions in Topotoolbox to find the best reference concavity, then which way can I find the best concavity, but I don't know if this reference concavity is the concavity or crack point of the ancient segment profile above Knickpoint The reference concavity of the river in the upstream drainage area, or the reference concavity of the river in the whole basin today.







------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Adam M. Forte"<notifications@github.com&gt;;
发送时间:&nbsp;2020年1月10日(星期五) 凌晨3:13
收件人:&nbsp;"amforte/Topographic-Analysis-Kit"<Topographic-Analysis-Kit@noreply.github.com&gt;;
抄送:&nbsp;"2621751827"<2621751827@qq.com&gt;;"Mention"<mention@noreply.github.com&gt;;
主题:&nbsp;Re: [amforte/Topographic-Analysis-Kit] Beginner's question (#36)




Hi @ZCG113035, for 1) no, ConditionDEM, RemoveFlats and FindThreshold are all optional for specific purposes. You can take the output of MakeStreams and it use it directly in other TAK functions.
 
For 2) I've been in the process of developing something like that, but I hope one can appreciate that I already have written a very large manual and that presenting case by case examples for ALL of the tools in TAK (Profiler was basically just my equivalent KsnProfiler) is a lot of effort (with me as the only developer) so it's taking a while.

**Comment by ZCG113035** [comment]
Thank you for your reply. Through this period of study, I can manually use most of the functions in TAK. TAK is very easy to use and powerful. Thank you for your contribution to this. I would like to ask how small the minarea should be. If it is 30 m pixels, it will usually be 10 ^ 6 in TAK, but it will be 10 ^ 3 in Topotoolbox, which is 0.9km ^ 2. I am currently dealing with the eastern Qinghai-Tibet Plateau Marginal DEM data, there is a deep-cut canyon area. The resolution of my DEM data is 145m. I also chose minarea as 10 ^ 6, and extracted the stream profile of interest based on manual selection, but the jagged profile appears. I suspect it is because of the quality of the data, but I have a few questions: 1) How big should minarea be, is it based on the DEM data resolution? 2) When doing Sementprojector, how much should the reference concavity be chosen to be 0.45 or 0.5 (in Choose between 0.4-0.6). So which way can I find the best concavity (Topotoolbox function or ChiProfile), but I do n’t know if this reference concavity is the concavity of the ancient segment profile above Knickpoint or the reference concavity of the river in the drainage area upstream above  Knickpoint. In other words, is the reference concavity of the reconstructed ancient channel profile representative of the relict landscape channel segment or intermediate landscape segment (if i find two knickpoints )or the whole channel today?

**Comment by amforte** [comment]
1) The minimum area is a proxy for where channels initiate. 1e6 m^2 is a commonly used criteria (regardless of data resolution, though obviously as your data resolution approaches 1 km pixels, this gets closer to 1 contributing pixel). In terms of what it should be, that depends a lot on the landscape in question. There are lots of papers on variety of thoughts about how to identify where channels initiate (either in the field or from DEM data). The 'FindThreshold' TAK function is designed to help users choose (or manually select) what might be a good threshold area for your region based on where it appears hillslopes (and debris dominated areas) transition to channels.

2) Similarly, the choice of reference concavity is very location dependent with a lot of literature describing what constitutes the 'right concavity'. The standard argument is that the natural variability on concavity is between 0.35-0.6 so any value in there is probably decent. The 0.45 value was popularized by Whipple and others, but there's nothing special about it. I mainly have 0.5 as a default as this makes the units on ksn m's (unit on ksn is m^2*reference concavity). Ultimately, as long as you're consistent with your reference concavity and it's a reasonable value, the choice is not incredibly important for most applications. The tools in TopoToolbox for finding appropriate concavities might be a good place to start. In general, when trying to find best fit concavities, you want to look at well adjusted streams (without knickpoints) because averaging across knickpoints will bias the concavity estimate.

As for jagged profiles, this is likely a function of data resolution. You could experiment with the ConditionDEM function to see if one of the algorithms there are able to smooth the profiles more, but ultimately, if the data is noisy, the smoothing will just give you a pretty result (it does not imply that it's any more 'true').

**Comment by ZCG113035** [comment]
Wow, I can receive your replies continuously after I ask questions continuously. I am really excited. 
1.I think now I'm probably clear that the 'minarea' you mentioned in TAK is the criteria area, which may represent the transition from debrisflow–dominated colluvial channels to stream-flow–dominated fluvial channels (wobus et al. 2006) instead of the code I saw in TopoToolbox (wschwanghart GitHub blog page): S = STREAMobj (FD, 'minarea', 500); where 500 is the number of pixels that determines 'minarea'.In other words, if the resolution is 10m, it represents an upslope area of 0.05 sqkm or 500 pixels. If the basin area is larger than this value, rivers will be generated？After listening to your 'FindThreshold' TAK function can help users choose (or manually select) what might be a good threshold area, but there are too many rivers that need me to choose, if running FindThreshold with 'num streams'set to'All', if ‘num streams’ is determined to be an exact number, I don’t know what it should be.
2. After listening to your explanation, I know that the reference concavity is 0.5, which means that the unit of ksn value is m ^ 2, just like when the reference concavity is 0.45, the unit of ksn is m ^ 0.9.
3.If I study in the following, is the SRTMDEM data better?
I am a graduate student from Nanjing Normal University in China. My name is chao gang Zheng. After hearing a report from Eric Kirby at the Institute of Geology, China Earthquake Administration in June 2019, I began to try to extract the river Knickpoint, the river profile, reconstruct stream profile to perform fault slip rate and rock uplift in my research area analysis and research.TAK is very powerful, but I found that most people around me use stream profile tools. But I think TAK is already practical for people like me who don't have much programming knowledge.I think if you will come to China for a lecture, I will definitely visit you. Thank you for your guidance these days.

**Comment by ZCG113035** [comment]
Wow, I can receive your replies continuously after I ask questions continuously. I am really excited.
1.I think now I'm probably clear that the 'minarea' you mentioned in TAK is the criteria area, which may represent the transition from debrisflow–dominated colluvial channels to stream-flow–dominated fluvial channels (wobus et al. 2006) instead of the code I saw in TopoToolbox (wschwanghart GitHub blog page): S = STREAMobj (FD, 'minarea', 500); where 500 is the number of pixels that determines 'minarea'.In other words, if the resolution is 10m, it represents an upslope area of 0.05 sqkm or 500 pixels. If the basin area is larger than this value, rivers will be generated？After listening to your 'FindThreshold' TAK function can help users choose (or manually select) what might be a good threshold area, but there are too many rivers that need me to choose, if running FindThreshold with 'num streams'set to'All', if ‘num streams’ is determined to be an exact number, I don’t know what it should be.
2. After listening to your explanation, I know that the reference concavity is 0.5, which means that the unit of ksn value is m ^ 2, just like when the reference concavity is 0.45, the unit of ksn is m ^ 0.9.
3.If I study in the following, is the SRTMDEM data better?
I am a graduate student from Nanjing Normal University in China. My name is chao gang Zheng. After hearing a report from Eric Kirby at the Institute of Geology, China Earthquake Administration in June 2019, I began to try to extract the river Knickpoint, the river profile, reconstruct stream profile to perform fault slip rate and rock uplift in my research area analysis and research.TAK is very powerful, but I found that most people around me use stream profile tools. But I think TAK is already practical for people like me who don't have much programming knowledge.I think if you will come to China for a lecture, I will definitely visit you. I think in the future, I need to strengthen the basic knowledge of programming, thank you for your guidance these days..
